src/quo2/components/user_avatar.cljs

```diff
@@ -0,0 +1,142 @@
+                    :status-indicator-border 2
+                    :font-size 13}
+            :xs {:outer 24
+                 :inner 24
+                 :status-indicator 0
+                 :status-indicator-border 0
+                 :font-size 13}
+            :xxs {:outer 20
+                  :inner 20
+                  :status-indicator 0
+                  :status-indicator-border 0
+                  :font-size 11}
+            :xxxs {:outer 16
+                   :inner 16
+                   :status-indicator 0
+                   :status-indicator-border 0
+                   :font-size 11}})
+
+(defn dot-indicator
+  [size status-indicator? online? ring? dark?]
+  (when status-indicator?
+    (let [dimensions (get-in sizes [size :status-indicator])
+          border-width (get-in sizes [size :status-indicator-border])
+          right (case size
+                  :big 2
+                  :medium 0
+                  :small 0
+                  0)
+          bottom (case size
+                   :big (if ring?
+                          4
+                          2)
+                   :medium (if ring?
+                             0
+                             -2)
+                   :small (if ring?
+                            0
+                            -2)
+                   0)]
+      [rn/view {:style {:background-color (if online?
+                                            colors/success-50
+                                            colors/neutral-40)
+                        :width dimensions
+                        :height dimensions
+                        :border-width border-width
+                        :border-radius dimensions
+                        :border-color (if dark?
+                                        colors/black
+                                        colors/white)
+                        :position :absolute
+                        :bottom bottom
+                        :right right}}])))
+
+(defn container-styling [inner-dimensions outer-dimensions]
+  {:width inner-dimensions
+   :position :absolute
+   :top (/ (- outer-dimensions inner-dimensions) 2)
+   :left (/ (- outer-dimensions inner-dimensions) 2)
+   :height inner-dimensions
+   :border-radius inner-dimensions})
+
+(defn container [inner-dimensions outer-dimensions & children]
+  (let [dark-kw (if dark? :dark :light)]
+    [rn/view {:style (merge {:background-color (colors/custom-color
+                                                :turquoise
+                                                dark-kw)
+                             :justify-content :center
+                             :align-items :center}
+                            (container-styling inner-dimensions outer-dimensions))}
+     children]))
+
+(defn user-avatar
+  [{:keys [ring?
+           online?
+           size
+           status-indicator?
+           profile-picture
+           full-name]
+    :or {full-name "john doe"
```

Comment:
better to have something like `"empty name"`

Reply:
Done